### PR TITLE
Adding isDebug flag to not send events when in debug mode

### PR DESCRIPTION
**Pull request by scalvert**
We need a debug flag in Hora so that while in debug mode you don't fire off events to GA. Instead, we log the data to the console, which allows us to inspect the values.

Status: **Ready to merge**

Reviewers: @ericmuyser 
JIRA: https://mobify.atlassian.net/browse/CSOPS-1110

## Changes
- added `Hora.isDebug` flag
- added parameter to `Hora.init` that accepts the debug flag and sets it
- added tests to verify behavior 

## How to test-drive this PR
- checkout this branch
- `grunt test`

### CODE REVIEW

File: src/js/hora.js [hunk 1]
```diff
@@ -634,7 +646,7 @@ define([
 
                 Hora.__stringifyPropertyValues(transaction);
 
-                Mobify.analytics.ua(ECOMMERCE_PLUGIN + ':addTransaction', transaction);
+                !Hora.isDebug && Mobify.analytics.ua(ECOMMERCE_PLUGIN + ':addTransaction', transaction);
```
Comment (ericmuyser) [inline src/js/hora.js]: You're making me wanna proxy Mobify.analytics.ua (like we do above) so we don't need these one-offs, haha.

**Comment by ericmuyser** [comment]
LGTM

File: src/js/hora.js (hunk 1 above)
Comment (scalvert) [inline src/js/hora.js]: No code changes to Hora after this!

Comment (ericmuyser) [inline src/js/hora.js]: Haha okay

**Comment by jnurse** [comment]
Looks good!

**Comment by scalvert** [comment]
I'll take that as a :+1:!

**Pull request merged**